File: longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = {}
        ans = 0
        lt = 0
        maxf = 0

        for rt in range(len(s)):
            count[s[rt]] = count.get(s[rt], 0) + 1
            maxf = max(maxf, count[s[rt]])
            
            while (rt-lt+1) - maxf > k:
                count[s[lt]] -= 1
                lt += 1

            ans = max(ans, rt-lt+1)

        return ans

File: longest-repeating-character-replacement.cpp
#include <string>
#include <unordered_map>

using namespace std;

class Solution {
public:
    int characterReplacement(string s, int k) {
		unordered_map<char, int> count;
		int ans = 0;
		int lt = 0;
		int maxf = 0;

		for (int rt=0; rt<s.size(); ++rt) {
			char ch = s[rt];
			count[ch] = (count.find(ch) != count.end())? count[ch]+1 : 1;
			maxf = max(maxf, count[ch]);
			
			while ((rt-lt+1) - maxf > k) {
				count[s[lt]] -= 1;
				lt += 1;
			}

			ans = max(ans, rt-lt+1);
		}

		return ans;
    }
};
